elastic/kibana
# [tsvb] read only mode

part of https://github.com/elastic/kibana/issues/154307

PR adds ability to put TSVB into read only mode - preventing TSVB visualizations from being created and edited.

To test:
* start kibana with `yarn start --serverless=es`
* add `vis_type_timeseries.readOnly: true` to kibana.yml

Visualization public plugin changes:
* Removes `hideTypes` from VisualizationSetup contract. Used by Maps plugin to set "hidden" to true for tile_map and region_map visualization types. In 8.0, tile_map and region_map visualization type registration moved into maps plugin so `hideTypes` no longer needed.
* Renamed vis type definition `hidden` to `disableCreate`.
* Added `disableEdit` to vis type definition.
* Hide edit link in dashboard panel options when `disableEdit` is true
* Does not display links and edit action in listing table when  `disableEdit` is true

Visualization server plugin changes:
* Add `readOnlyVisType` registry to set up contract
* Update visualization savedObject.management.getInAppUrl to return undefined when vis type has been registered as readOnly.
    * Prevents "readOnly "visualization types from being displayed in global search results
    * Prevents "readOnly "visualization types from having links in saved object management listing table.

Timeseries server plugin changes:
* Add `readOnly` yaml configuration
* Expose `readOnly` yaml configuration to public
* When `readOnly` is true, call VisualizationsServerSetup.registerReadOnlyVisType to mark vis type as read only

Timeseries public plugin changes:
* Set disableCreate and disableEdit to true when `readOnly` is true
